CMakeLists.txt

```diff
@@ -19,6 +19,64 @@ if (NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
             "MinSizeRel" "RelWithDebInfo")
 endif ()
 
+# check and set compiler 
+set(REQUIRED_GNU_VERSION 13.0.0)
+set(REQUIRED_CLANG_VERSION 17.0.0)
+
+find_program(GPP_EXECUTABLE NAMES g++)
+find_program(CLANG_EXECUTABLE NAMES clang++)
+
+set(GPP_VERSION "0.0.0")
+set(CLANG_VERSION "0.0.0")
+
+if(GPP_EXECUTABLE)
+    execute_process(
+        COMMAND ${GPP_EXECUTABLE} -dumpversion
+        OUTPUT_VARIABLE GPP_VERSION
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+endif()
+
+if(CLANG_EXECUTABLE)
+    execute_process(
+        COMMAND ${CLANG_EXECUTABLE} --version
+        OUTPUT_VARIABLE CLANG_VERSION_OUTPUT
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+    string(REGEX MATCH "[0-9]+\\.[0-9]+\\.[0-9]+" CLANG_VERSION ${CLANG_VERSION_OUTPUT})
+endif()
+
+if(GPP_EXECUTABLE AND GPP_VERSION VERSION_GREATER_EQUAL REQUIRED_GNU_VERSION)
+    set(CMAKE_CXX_COMPILER ${GPP_EXECUTABLE})
+    message(STATUS "Using GNU ${GPP_VERSION} for all modules except cgroup_dev_bpf_object")
+
+    if(CLANG_EXECUTABLE)
+        if(CLANG_VERSION VERSION_GREATER_EQUAL REQUIRED_CLANG_VERSION)
+            set(ENABLE_BPF ON)
+            message(STATUS "Using Clang ${CLANG_VERSION} for cgroup_dev_bpf_object module")
+        else()
+            set(ENABLE_BPF OFF)
+            message(WARNING "Clang ${CLANG_VERSION} or higher is required for cgroup_dev_bpf_object module. Disable cgroup_dev_bpf_object module")
```

Comment:
Clang ${CLANG_VERSION} or higher is required for device management on Cgroup V2, disabling it . You can use Cgroup V1.